presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -169,6 +169,14 @@ private void validateShowStats(ShowStats node)
                 check(query.getQueryBody() instanceof QuerySpecification, node, "Only table and simple table subquery can be passed as argument to SHOW STATS clause");
                 QuerySpecification querySpecification = (QuerySpecification) query.getQueryBody();
 
+                Plan plan = queryExplainer.get().getLogicalPlan(session, new Query(Optional.empty(), querySpecification, Optional.empty(), Optional.empty()), parameters);
+
+                Optional<FilterNode> filterNode = searchFrom(plan.getRoot())
+                        .where(FilterNode.class::isInstance)
+                        .findSingle();
+
+                check(!filterNode.isPresent(), node, "Only columns that can be pushed down are supported in SHOW STATS WHERE clause");
```

Comment:
Change the message to say `Only predicates that can be pushed down are supported in the SHOW STATS WHERE clause`